plugins/ltac2/tac2tactics.ml

```diff
@@ -456,3 +456,21 @@ let inversion knd arg pat ids =
 let contradiction c =
   let c = Option.map mk_with_bindings c in
   Contradiction.contradiction c
+
+let congruence n l = 
+  let n = match n with
+  | None -> 1000
```

Comment:
It's not very nice to have this magic number in multiple places, how about modifying the cctac API to take `int option` and handle the default value internally instead?

Reply:
I've done it because in the Ltac implementation they are hardcoded into the mlg file. Should I change that, so that they are hardcoded into cctac.ml and change the mlg file to call congruence_tac with option types?

Reply:
That is what I suggest

Reply:
I changed it this way